builder/ebssurrogate/builder_test.go

```diff
@@ -57,6 +57,67 @@ func TestBuilderPrepare_InvalidKey(t *testing.T) {
 	}
 }
 
+func TestBuilderPrepare_UefiData(t *testing.T) {
+	var b Builder
+	// Basic configuration
+	b.config.RootDevice = RootBlockDevice{
+		SourceDeviceName: "device name",
+		DeviceName:       "device name",
+	}
+	b.config.LaunchMappings = BlockDevices{
+		BlockDevice{
+			BlockDevice: common.BlockDevice{
+				DeviceName: "device name",
+			},
+			OmitFromArtifact: false,
+		},
+	}
+	b.config.AMIVirtType = "type"
+	config := testConfig()
+	config["ami_name"] = "name"
+
+	// Test bad
+	config["uefi_data"] = "foo"
```

Comment:
As with the chroot tests, I think this is a good candidate for subtests with a common function for executing them.